crowbar_framework/app/models/deployer_service.rb

```diff
@@ -181,6 +160,23 @@ def transition(inst, name, state)
       roles << node.run_list_to_roles
       roles.flatten!
 
+      # Allocate required addresses
+      range = node.admin? ? "admin" : "host"
+      @logger.debug("Deployer transition: Allocate admin address for #{name}")
+      ns = NetworkService.new @logger
+      result = ns.allocate_ip("default", "admin", range, name)
+      @logger.error("Failed to allocate admin address for: #{node.name}: #{result[0]}") if result[0] != 200
+      if result[0] == 200
+        address = result[1]["address"]
+        boot_ip_hex = sprintf("%08X", address.split(".").inject(0){ |acc,i| (acc << 8) + i.to_i })
+        # save this on the node after it's been refreshed with the network info.
+        node.crowbar["crowbar"]["boot_ip_hex"] = boot_ip_hex if boot_ip_hex
+      end
+
+      @logger.debug("Deployer transition: Done Allocate admin address for #{name} boot file:#{boot_ip_hex}")
+
+      save_it = true
```

Comment:
Lint/UselessAssignment: Useless assignment to variable - `save_it`. (https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)